lib/models/environment-variables.coffee

```diff
@@ -158,22 +159,23 @@ exports.device = {}
 # @public
 # @function
 #
-# @param {(String|Number)} deviceId - device id
+# @param {String} deviceName - device name
 # @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
 #
 # @example
-#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#	resin.models.environmentVariables.device.getAll 'MyDevice', (error, environmentVariables) ->
 #		throw error if error?
 #		console.log(environmentVariables)
 ###
-exports.device.getAll = (deviceId, callback) ->
-	return pine.get
-		resource: 'device_environment_variable'
-		options:
-			filter:
-				device: deviceId
-			expand: 'device'
-			orderby: 'env_var_name asc'
+exports.device.getAll = (deviceName, callback) ->
+	deviceModel.get(deviceName).then (device) ->
```

Comment:
The idea is cool, however I think it makes there interface a little bit complex, we will have to support this in the rest of the functions (which would be a major change) and I guess users will not bother to select the exact things they need (and it implies the users know the name of the property in advance), and therefore will be a feature we will be using only internally. 

I suggest implementing a custom function that only returns the id (we can keep it private as well), or wait until the SDK becomes too network consuming, so we don't optimise prematurely.